packages/playground/php-cors-proxy/cors-proxy-functions.php

```diff
@@ -354,3 +354,24 @@ function rewrite_relative_redirect(
     }
     return $proxy_absolute_url . $redirect_location;
 }
+
+/**
+ * Answers whether CORS is allowed for the specified origin.
+ */
+function should_respond_with_cors_headers($host, $origin) {
+    if (
+        $host !== 'playground.wordpress.net' &&
```

Comment:
Should we also refuse to serve requests coming from playground.wordpress.net?

Reply:
@adamziel do you mean requests from the current playground.wordpress.net proxy server to the dedicated proxy server?

Maybe it would be good for us to send a custom user agent like `WordPressPlaygroundCorsProxy`. We could deny all requests with the `WordPressPlaygroundCorsProxy` user agent. And it would probably be good to be as forthright as possible with the target servers about the nature of our requests.

Maybe we can make this a configurable header and only set it for the proxy deployed to WP cloud. That way, folks deploying their own instances of the proxy (which won't have most of the defensive meaures) won't send that user agent by default.

What do you think?

Reply:
I meant making sure we can't request `https://playground.wordpress.net/cors-proxy.php` – mu assumption was that the script would still be colocated with the website deployment, which may not be true at all.